src/Apps/Auction/Components/RegistrationForm.tsx

```diff
@@ -260,41 +280,70 @@ const RegistrationForm: React.FC<RegistrationFormProps> = props => {
     })
   }
 
-  return (
-    <Formik
-      initialValues={{
-        name: "",
-        street1: "",
-        city: "",
-        credit_card: null,
-        state: "",
-        postal_code: "",
-        telephone: "",
-        agree_to_terms: false,
-      }}
-      onSubmit={(values: FormValues, { setFieldError, setSubmitting }) => {
-        const address = {
-          name: values.name,
-          address_line1: values.street1,
-          address_city: values.city,
-          address_state: values.state,
-          address_zip: values.postal_code,
-        }
+  const [showErrorModal, setShowErrorModal] = useState(false)
 
-        props.stripe.createToken(address).then(({ error, token }) => {
-          if (error) {
-            setFieldError("credit_card", error.message)
-            setSubmitting(false)
-          } else {
-            createCreditCard(token.id).then(() => {
-              createBidder(setSubmitting)
-            })
+  return (
+    <>
+      <Formik
+        initialValues={{
+          name: "",
+          street1: "",
+          country: "",
+          city: "",
+          credit_card: null,
+          state: "",
+          postal_code: "",
+          telephone: "",
+          agree_to_terms: false,
+        }}
+        onSubmit={(values: FormValues, actions) => {
+          const address = {
+            name: values.name,
+            address_line1: values.street1,
+            address_country: values.country,
+            address_city: values.city,
+            address_state: values.state,
+            address_zip: values.postal_code,
           }
-        })
-      }}
-      validationSchema={validationSchema}
-      render={InnerForm}
-    />
+
+          const { setFieldError, setSubmitting } = actions
+          const { sale } = props
+
+          props.stripe.createToken(address).then(({ error, token }) => {
+            if (error) {
+              setFieldError("credit_card", error.message)
+              setSubmitting(false)
+            } else {
+              if (props.onSubmit) {
```

Comment:
can you elaborate on this? When wouldn't `props.onSubmit` be true?

Reply:
This is intended to override the default onSubmit behavior for use in storybooks or tests. @erikdstock and I have been thinking of some component reorganization that would make this required on the form, moving some of the onSubmit behavior up a level.